spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -371,16 +372,12 @@ public void deleteWhere(Predicate[] predicates) {
         icebergTable
             .newDelete()
             .set("spark.app.id", sparkSession().sparkContext().applicationId())
+            .toBranch(
+                SparkTableUtil.wapEnabled(table())
+                    ? SparkTableUtil.determineWriteBranch(sparkSession(), branch)
+                    : branch())
             .deleteFromRowFilter(deleteExpr);
 
-    if (SparkTableUtil.wapEnabled(table())) {
-      branch = SparkTableUtil.determineWriteBranch(sparkSession(), branch);
-    }
-
-    if (branch != null) {
-      deleteFiles.toBranch(branch);
-    }
```

Comment:
Does this need to change? Like the situation above, I'd prefer not to call `toBranch` with `main`.